gnss_lib_py/utils/gnss_models.py

```diff
@@ -0,0 +1,658 @@
+"""Model GNSS measurements and measurement corrections.
+
+Functions to generate expected measurements, simulate noisy mesurements,
+and estimate pseudorange corrections (clock corrections, ionospheric and
+tropospheric delays) for given receiver states and time.
+"""
+
+__authors__ = "Ashwin Kanhere, Bradley Collicott"
+__date__ = "17 Jan, 2023"
+
+import warnings
+
+import numpy as np
+from numpy.random import default_rng
+
+import gnss_lib_py.utils.constants as consts
+from gnss_lib_py.utils.coordinates import ecef_to_geodetic, ecef_to_el_az
+from gnss_lib_py.parsers.navdata import NavData
+from gnss_lib_py.utils.time_conversions import gps_millis_to_tow
+from gnss_lib_py.utils.sv_models import _find_visible_ephem, _extract_pos_vel_arr, \
+                        _find_sv_location, find_sv_states, _compute_eccentric_anomaly, \
+                        _find_visible_sv_posvel, _sort_ephem_measures, \
+                        _filter_ephemeris_measurements
+
+
+def add_measures(measurements, ephemeris_path, iono_params=None,
+                 pseudorange=True, doppler=True, corrections=True,
+                 delta_t_dec = -2):
+    """Estimate measurements and add to given measurements
```

Comment:
Maybe add which measurements are being estimated to preamble? Might be unclear for those unfamiliar with the library